File: Linked_list/LinkedList.java
import java.io.*;

public class LinkedList {

  Node head;  // head of the linked list

  // linked list node
  static class Node { // inner class is static to allow main() to access it
    double val;
    Node next;

    // constructor
    Node(double v) {
      val = v;
      next = null;
    }
  }

  public static LinkedList insert(LinkedList list, double val) {
    Node newNode = new Node(val);

    if(list.head == null) list.head = newNode;
    else {
      Node last = list.head;
      while (last.next != null) last = last.next;
      last.next = newNode;
    }

    return list;
  }

  public static boolean isMember(LinkedList list, double val) {
    boolean membership = false;
    boolean runSearch = true;
    Node currentNode = list.head;

    if(currentNode.val == val) membership = true;
    while (runSearch && !membership) {
      if(currentNode.val == val) membership = true;
      else {
        if(currentNode.next != null) currentNode = currentNode.next;
        else runSearch = false;
      }
    }

    return membership;
  }

  public static void main(String[] args) {
    LinkedList myList = new LinkedList();
    double searchVal = 7.28;

    myList = insert(myList, 7.28);
    myList = insert(myList, 1.23);
    myList = insert(myList, 4.56);
    myList = insert(myList, 7.89);
    myList = insert(myList, 6.42);
    myList = insert(myList, 9.63);

    if(isMember(myList, searchVal)) System.out.println(searchVal + " is a member.");
    else System.out.println(searchVal + " is NOT a member.");
  }
}

File: Linked_list/linked_list.cpp
// This program is an exercise in linked list use. Normally, we would use the STL to implement a list, but to make sure we understand pointer variables, we're making our own from scratch.

#include <iostream>

using namespace std;

struct node{
    double val;
    node *next = NULL;  // pointer to the next node
};

class LinkedList {
private:
  node *head;
  node *last;
public:
  void add(double x)
  {
      node *newNode = new node; // declare new node, assign value and address of next node
      newNode->val = x;
      newNode->next = NULL;
      if(!head){                // if the linked list has no head node
        head = newNode;         // the address for the head node is that of the new node
        last = newNode;         // the address for the last node is that of the new node
      }
      else                      // if a head node already exists
      {
        last->next = newNode;   // set the current last node's next pointer to the address of the new node
        last = newNode;         // make the new node the new last node
      }
  }

  bool isMember(double x)
  {
    bool membership = false;
    bool runSearch = true;
    node *currentNode = new node;
    currentNode = head;

    // check if head node contains the searched value
    if(currentNode->val == x) membership = true;
    while(runSearch && !membership)
    {
      // finding the searched value ends the search
      if(currentNode->val == x) membership = true;
      else
      {
        // if search value not found, advance to the next node if one exists
        if(currentNode->next) currentNode = currentNode->next;
        // if no next node exists, terminate the search
        else runSearch = false;
      }
    }

    return membership;
  }

  // default constructor
  LinkedList()
  {
    head = NULL;
    last = NULL;
  }
};

int main()
{
  LinkedList linkList;
  double searchVal = 7.28;

  linkList.add(5);
  linkList.add(12385);
  linkList.add(7.28);
  if(linkList.isMember(searchVal)) cout << searchVal << " is a member." << endl;
  else cout << searchVal << " is not a member." << endl;

  return 0;
}
